onnxruntime/test/optimizer/nchwc_optimizer_test.cc

```diff
@@ -0,0 +1,839 @@
+  }
+
+  NodeArg* MakeIntermediate() {
+    std::string name = graph_.GenerateNodeArgName("node");
+    return &graph_.GetOrCreateNodeArg(name, nullptr);
+  }
+
+  NodeArg* MakeInitializer(const std::vector<int64_t>& shape) {
+    std::string name = graph_.GenerateNodeArgName("constant");
+    ONNX_NAMESPACE::TensorProto tensor_proto;
+    tensor_proto.set_name(name);
+    tensor_proto.set_data_type(ONNX_NAMESPACE::TensorProto_DataType_FLOAT);
+
+    int64_t num_elements = 1;
+    for (auto& dim : shape) {
+      tensor_proto.add_dims(dim);
+      num_elements *= dim;
+    }
+
+    auto random_data = FillRandomData(static_cast<size_t>(num_elements));
+    tensor_proto.mutable_float_data()->Resize(static_cast<int>(num_elements), 0.0f);
+    memcpy(tensor_proto.mutable_float_data()->mutable_data(), random_data.data(), random_data.size() * sizeof(float));
+
+    graph_.AddInitializedTensor(tensor_proto);
+
+    return &graph_.GetOrCreateNodeArg(name, nullptr);
+  }
+
+  Node& AddNode(const std::string& op_type,
+                const std::vector<NodeArg*>& input_args,
+                const std::vector<NodeArg*>& output_args) {
+    return graph_.AddNode(graph_.GenerateNodeName("node"),
+                          op_type,
+                          "description",
+                          input_args,
+                          output_args);
+  }
+
+  Node& AddConvNode(NodeArg* input_arg, NodeArg* output_arg, const std::vector<int64_t>& weights_shape) {
+    auto* weights_arg = MakeInitializer(weights_shape);
+    auto* biases_arg = MakeInitializer({weights_shape[0]});
+    return AddNode("Conv", {input_arg, weights_arg, biases_arg}, {output_arg});
+  }
+
+  std::vector<float> FillRandomData(size_t count) {
+    constexpr int min_fill_value = -23;
+    constexpr int max_fill_value = 23;
+
+    std::vector<float> random_data;
+    random_data.resize(count);
+    for (size_t n = 0; n < count; n++) {
+      random_data[n] = static_cast<float>(fill_value_);
+      fill_value_++;
+      if (fill_value_ == max_fill_value) {
+        fill_value_ = min_fill_value;
+      }
+    }
+    return random_data;
+  }
+
+  Graph& graph_;
+  NameMLValMap feeds_;
+  std::vector<std::string> output_names_;
+  int fill_value_;
+};
+
+void NchwcOptimizerTester(const std::function<void(NchwcTestHelper& helper)>& build_test_case,
+                          const std::function<void(NchwcInferenceSession& session)>& check_nchwc_graph) {
+  // Ignore the test if NCHWc is not supported by the platform.
+  if (MlasNchwcGetBlockSize() <= 1) {
+    return;
+  }
+
+  // Build the model for this test.
+  Model model("nchwc");
+  NchwcTestHelper helper(model.MainGraph());
+  build_test_case(helper);
+  ASSERT_TRUE(model.MainGraph().Resolve().IsOK());
+
+  // Serialize the model to a string.
+  std::string model_data;
+  model.ToProto().SerializeToString(&model_data);
+
+  auto run_model = [&](NchwcTestHelper& helper, TransformerLevel level, std::vector<OrtValue>& fetches) {
```

Comment:
'helper' is captured by reference so doesn't need to be passed in

Reply:
Fixed, thanks.